microsoft/testsuites/dpdk/dpdkvpp.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import time
+from typing import List, Type
+
+from assertpy import assert_that
+
+from lisa.executable import Tool
+from lisa.operating_system import Fedora, Ubuntu
+from lisa.tools import Gcc, Git, Make
+from lisa.util import UnsupportedDistroException
+
+
+class DpdkVpp(Tool):
+
+    VPP_SRC_LINK = "https://github.com/FDio/vpp.git"
+    REPO_DIR = "vpp"
+
+    @property
+    def command(self) -> str:
+        return "vpp"
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Gcc, Make, Git]
+
+    def _install(self) -> bool:
+        node = self.node
+        if isinstance(node.os, Ubuntu):
+            node.os.add_repository("ppa:canonical-server/server-backports")
+            pkg_type = "deb"
+        elif isinstance(node.os, Fedora):
+            node.os.install_epel()
+            pkg_type = "rpm"
+        else:
+            raise UnsupportedDistroException(self.node.os)
+
+        node.execute(
+            (
+                "curl -s https://packagecloud.io/install/repositories/fdio/release/"
+                f"script.{pkg_type}.sh | sudo bash"
+            ),
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "Could not install vpp with fdio provided installer"
+            ),
+        )
+        node.os.update_packages("")
+        return True
+
+    def install_from_package_manager(self) -> None:
```

Comment:
It looks always called after the install. What's the reason to make it as a separated public method?

Reply:
Just breaking up the longer running items. I'll change it so it's a private method called from install.